File: GoService/info/info.go
package info

import (
	"GoService/config"
	"GoService/handlers"
)

var Config = config.LoadConfig("")

type Info struct {
	Name 				string 						`json:"name"`
	Tags 				[]string 					`json:"tags"`
	HashName			string						`json:"hashname"`
	BashType 			string 						`json:"bashtype"`
	UseRedis			bool						`json:"useredis"`
	RedisConnection 	config.RedisConnectionData	`json:"redisconnection"`
	SignsNumber			int							`json:"signsnumber"`
	ContainersNumber	int							`json:"containersnumber"`
	ConnectionData 		config.ConnectionData 		`json:"connectiondata"`
}

func GetInfo() Info {
	var info Info

	name := Config.Name
	tags := Config.Tags
	hashname := Config.HashName
	bashtype := Config.BashType
	useredis := Config.UseRedis
	redisconnection := Config.RedisConnection
	signsnumber, _ := handlers.SignsNumber("")
	containersnumber, _ := handlers.ContainersNumber()

	info = Info {
		Name: name,
		Tags: tags,
		HashName: hashname,
		BashType: bashtype,
		UseRedis: useredis,
		RedisConnection: redisconnection,
		SignsNumber: signsnumber,
		ContainersNumber: containersnumber,
	}

	return info
}

File: frontend/src/structures/Info.ts
export interface RedisConnectionData {
  host: string;
  port: string;
  password: string;
  db: number;
}

const defaultRedisConnection = {
  host: "",
  port: "",
  password: "",
  db: 0,
};

export interface ConnectionData {
  host: string;
  port: string;
}

const defaultConnectionData = {
  host: "",
  port: "",
};

export interface Info {
  name: string;
  tags: string[];
  hashname: string;
  bashtype: string;
  useredis: boolean;
  redisconnection: RedisConnectionData;
  signsnumber: number;
  containersnumber: number;
  connectiondata: ConnectionData;
}

export const defaultInfo = {
  name: "",
  tags: [],
  hashname: "",
  bashtype: "",
  useredis: false,
  redisconnection: defaultRedisConnection,
  signsnumber: 0,
  containersnumber: 0,
  connectiondata: defaultConnectionData,
};
